### PR TITLE
intel: Add driver for QSPI

### PR DESCRIPTION
**Pull request by hadi-asyrafi**
To support the enablement of QSPI booting

Signed-off-by: Muhammad Hadi Asyrafi Abdul Halim <muhammad.hadi.asyrafi.abdul.halim@intel.com>

### CODE REVIEW

File: plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c [hunk 1]
```diff
@@ -0,0 +1,823 @@
...
+
+	status = cad_qspi_stig_read_cmd(CAD_QSPI_STIG_OPCODE_RDID, 0, 3,
+					&rdid);
+
+	if (status != 0) {
+		ERROR("Error reading RDID\n");
+		return status;
+	}
+
+	/*NOTE: The Size code seems to be a form of BCD (binary coded decimal).
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c]: nit: Leave the first line empty. Also, leave a space between * and the text.

File: plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.h [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+#define CAD_QSPI_DEVSZ_BYTES_PER_BLOCK(x)		(x << 16)
+
+#define CAD_QSPI_DEVWR					0x8
+#define CAD_QSPI_DEVRD					0x4
+#define CAD_QSPI_DEV_OPCODE(x)				((x & 0xff) << 0)
+#define CAD_QSPI_DEV_INST_TYPE(x)			((x & 0x3) << 8)
+#define CAD_QSPI_DEV_ADDR_TYPE(x)			((x & 0x3) << 12)
+#define CAD_QSPI_DEV_DATA_TYPE(x)			((x & 0x3) << 16)
+#define CAD_QSPI_DEV_MODE_BIT(x)			((x & 0x1) << 20)
+#define CAD_QSPI_DEV_DUMMY_CLK_CYCLE(x)			((x & 0xf) << 24)
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.h]: Add parentheses to the macro arguments. There are several occurrences more in this file.

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

File: plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.h (hunk 2 above)
Comment (hadi-asyrafi) [reply plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.h]: Thanks for the mention =D

File: plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c (hunk 1 above)
Comment (hadi-asyrafi) [reply plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c]: Noted & thanks

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
Some checkpatch problems:
```
plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c:0: ERROR: do not set execute permissions for source files

plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.c:533: WARNING: 'diable' may be misspelled - perhaps 'disable'?
+		ERROR("failed diable\n");

plat/intel/soc/stratix10/drivers/qspi/cadence_qspi.h:824: ERROR: do not set execute permissions for source files
```

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Pull request merged**